source/reference/operator/aggregation/binarySize.txt

```diff
@@ -0,0 +1,232 @@
+     { _id: 5, name: "empty.jpg", binary: new BinData(0, "") }
+   ])
+
+The following aggregation :pipeline:`projects <$project>`:
+
+- The ``name`` field
+
+- The ``imageSize`` field, which uses :expression:`$binarySize` to
+  return the size of the document's ``binary`` field in bytes.
+
+.. code-block:: javascript
+
+   db.images.aggregate([
+     {
+       $project: {
+         "name": "$name",
+         "imageSize": { $binarySize: "$binary" }
+       }
+     }
+   ])
+
+The operation returns the following result:
+
+.. code-block:: javascript
+   :copyable: false
+
+   { "_id" : 1, "name" : "cat.jpg", "imageSize" : 16 }
+   { "_id" : 2, "name" : "big_ben.jpg", "imageSize" : 41 }
+   { "_id" : 3, "name" : "teaset.jpg", "imageSize" : 16 }
+   { "_id" : 4, "name" : "concert.jpg", "imageSize" : 269 }
+   { "_id" : 5, "name" : "empty.jpg", "imageSize" : 0 }
+
+Find Largest Binary Data
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following pipeline returns the image with the largest 
+binary data size:
+
+.. code-block:: javascript
+   
+   db.images.aggregate([
+      // First Stage
+      { $project: { name: "$name", imageSize: { $binarySize: "$binary" } }  },
+      // Second Stage
+      { $sort: { "imageSize" : -1 } },
+      // Third Stage
+      { $limit: 1 }
+   ])
+
+First Stage
+  The first stage of the pipeline :pipeline:`projects <$project>`:
+  
+  - The ``name`` field
+
+  - The ``imageSize`` field, which uses :expression:`binarySize` to
+    return the size of the document's ``binary`` field in bytes.
+
+  This stage outputs the following documents to the next stage:
+
+  .. code-block:: javascript
+     :copyable: false
+
+     { "_id" : 1, "name" : "cat.jpg", "imageSize" : 16 }
+     { "_id" : 2, "name" : "big_ben.jpg", "imageSize" : 41 }
+     { "_id" : 3, "name" : "teaset.jpg", "imageSize" : 16 }
+     { "_id" : 4, "name" : "concert.jpg", "imageSize" : 269 }
+     { "_id" : 5, "name" : "empty.jpg", "imageSize" : 0 }
+
+Second Stage
+  The second stage :pipeline:`sorts <$sort>` the documents by
+  ``imageSize`` in descending order.
+
+  This stage outputs the following documents to the next stage:
+
+  .. code-block:: javascript
+     :copyable: false
+
+     { "_id" : 4, "name" : "concert.jpg", "imageSize" : 269 }
+     { "_id" : 2, "name" : "big_ben.jpg", "imageSize" : 41 }
+     { "_id" : 1, "name" : "cat.jpg", "imageSize" : 16 }
+     { "_id" : 3, "name" : "teaset.jpg", "imageSize" : 16 }
+     { "_id" : 5, "name" : "empty.jpg", "imageSize" : 0 }
+
+Third Stage
+  The third stage :pipeline:`limits <$limit>` the output documents to
+  only return the document appearing first in the sort order:
+
+  .. code-block:: javascript
+     :copyable: false
+
+     { "_id" : 4, "name" : "concert.jpg", "imageSize" : 269 }
+
+.. seealso:: 
+
+   - :pipeline:`$project`
+   - :pipeline:`$sort`
+   - :pipeline:`$limit`
```

Comment:
Can we add reference to $strLenBytes  to $binarySize and $binarySize to $strLenBytes pages?

That way we can help people know there is another operator and not have to have people memorize what operators are available.

Reply:
Oh, and add to redirects and to the release notes/4.4.txt